models/mnist_gan_graph.py

```diff
@@ -0,0 +1,232 @@
+        d_fake = discriminator(g_sample, reuse=True)
+
+        d_loss_real = tf.reduce_mean(
+            tf.nn.sigmoid_cross_entropy_with_logits(logits=d_real, labels=tf.ones_like(d_real))
+        )
+        d_loss_fake = tf.reduce_mean(
+            tf.nn.sigmoid_cross_entropy_with_logits(logits=d_fake, labels=tf.zeros_like(d_fake))
+        )
+        self.d_loss = d_loss_real + d_loss_fake
+        self.g_loss = tf.reduce_mean(
+            tf.nn.sigmoid_cross_entropy_with_logits(logits=d_fake, labels=tf.ones_like(d_fake))
+        )
+
+        # Test summaries
+        tiled_image_random = tile_images(g_sample, 6, 6, 28, 28)
+        tiled_image_interpolated = tile_images(generator(self.noise_input_interpolated, reuse=True), 6, 6, 28, 28)
+        with self.summary_writer.as_default(), tf_summary.always_record_summaries():
+            gen_image_summary_op = tf_summary.image(
+                'generated_images', tiled_image_random, max_images=1, step=self.g_step
+            )
+            gen_image_summary_interpolated_op = tf_summary.image(
+                'generated_images_interpolated', tiled_image_interpolated, max_images=1, step=self.g_step
+            )
+            self.IMAGE_SUMMARIES = [gen_image_summary_op, gen_image_summary_interpolated_op]
+
+        # Optimizers
+        t_vars = tf.trainable_variables()
+        self.d_opt = tf.train.AdamOptimizer(2e-4).minimize(
+            self.d_loss, var_list=[var for var in t_vars if 'Discriminator' in var.name], global_step=self.d_step
+        )
+        self.g_opt = tf.train.AdamOptimizer(2e-4).minimize(
+            self.g_loss, var_list=[var for var in t_vars if 'Generator' in var.name], global_step=self.g_step
+        )
+
+        # saver = tf.train.Saver(max_to_keep=1)
+
+    def load_data(self):
+        # Read the input data
+        # In this case, MNIST + batch and shuffle it. In our case, it will be quite different.
+
+        from tensorflow.examples.tutorials.mnist import input_data
+
+        def read_data_sets(data_dir):
+            """Returns training and test tf.data.Dataset objects."""
+            data = input_data.read_data_sets(data_dir, one_hot=True)
+            # train_ds = tf.data.Dataset.from_tensor_slices((data.train.images,
+            #                                               data.train.labels))
+            # test_ds = tf.data.Dataset.from_tensors(
+            #   (data.test.images, data.test.labels))
+            return (data.train, data.test)
+
+        device, data_format = ('/gpu:0', 'channels_first')
+        if FLAGS.training.no_gpu:
+            device, data_format = ('/cpu:0', 'channels_last')
+        print('Using device %s, and data format %s.' % (device, data_format))
+
+        # Load the datasets
+        train_ds, test_ds = read_data_sets(FLAGS.data.in_dir)
+        # train_ds = train_ds.shuffle(60000).batch(FLAGS.batch_size)
+        return shuffle(train_ds.images, train_ds.labels)
+
+    def train_batch(self, batch):
```

Comment:
Please add a TODO checkpoint to check the tf.contrib.Dataset class. It might after some initial struggling make our life easier.

Reply:
Not adding the comments, added issue #5.